synapse/handlers/sync.py

```diff
@@ -690,6 +791,12 @@ def _generate_sync_entry_for_rooms(self, sync_result_builder, account_data_by_ro
 
             tags_by_room = yield self.store.get_tags_for_user(user_id)
 
+        yield self._update_room_entries_for_paginated_sync(
```

Comment:
I might be missing something - but I don't think this is sufficient or an effective way to breakdown sync - since which room is synced is still not granular. The client *should* really be able to specify which room should be synced first, otherwise it's going to waste time syncing rooms which the user might not be currently interested in, no?